### PR TITLE
Make deprecations versions explicit

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

Messages without `%(since)s` and `%(removal)s` will be ambiguous as to when they were deprecated and when they will be removed.

Also clean up extra 'in's in the messages as `%(removal)s` will add the correct preposition.

## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [N/A] New features are documented, with examples if plot related
- [N/A] Documentation is sphinx and numpydoc compliant
- [N/A] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [N/A] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by anntzer** [comment]
I wonder whether we should just make the deprecation machinery check the contents of `message` and, if it doesn't contain `%(since)s` or `%(removal)s`, append something like "This deprecation started in mpl %(since)s and will elapse %(removal)s"?

**Comment by QuLogic** [comment]
I was thinking about that, but it seems like one or two intentionally left it out; maybe we should only do that if `pending=False`?

**Comment by QuLogic** [comment]
Marking as RC, as we shouldn't put out a release without the right numbers, whatever way we decide to do it.

**Comment by anntzer** [comment]
pending=False explicitly disallows having a scheduled removal, so sure.

File: lib/matplotlib/colors.py [hunk 1]
```diff
@@ -1255,8 +1256,9 @@ def __init__(self, linthresh, linscale=1.0, vmin=None, vmax=None,
         if base is None:
             self._base = np.e
             cbook.warn_deprecated(
-                "3.3", message="default base will change from np.e to 10.  To "
-                "suppress this warning specify the base kwarg.")
+                "3.2", removal="3.3", message="default base will change from "
```
Comment (tacaswell) [inline lib/matplotlib/colors.py:1259]: I'm worried that the wall time on this is too fast.  Can we push it to 3.4?

**Comment by QuLogic** [comment]
I bumped the dates for the above `to_rgba_array` and #16417.

**Comment by tacaswell** [comment]
Going to make an executive decision and merge this.

**Pull request merged**